lib/model/message_list.dart

```diff
@@ -155,6 +155,30 @@ mixin _MessageSequence {
     _processMessage(messages.length - 1);
   }
 
+  /// Removes the given messages, if present.
+  ///
+  /// Returns true if at least one message was present, false otherwise.
+  /// If none of [messageIds] are found, this is a no-op.
+  bool _removeMessagesById(Iterable<int> messageIds) {
+    final messagesToRemoveById = <int>{};
+    final contentToRemove = <ZulipContent>{};
```

Comment:
```suggestion
    final contentToRemove = Set<ZulipContent>.identity();
```

That has no direct effect, because `ZulipContent` doesn't override `==` or `hashCode`. But it's not totally obvious that it wouldn't — indeed some of the other `ContentNode` subclasses do — so best to be explicit.

And comparing for identity, rather than general equality, is crucial to this algorithm's behavior both for performance and, in edge cases, for correctness.

Reply:
Ah that's interesting. So an identity set uses [identical](https://api.dart.dev/stable/3.4.4/dart-core/identical.html) to distinguish elements, which compares the object references instead.